components/TextFields/src/private/MDCTextInputCommonFundament.m

```diff
@@ -534,20 +533,32 @@ - (void)updateClearButton {
 }
 
 - (void)updateClearButtonConstraints {
-  CGFloat constant = MDCTextInputClearButtonImageSquareWidthHeight * [self clearButtonAlpha];
-  if (self.clearButtonWidth.constant != constant) {
-    self.clearButtonWidth.constant = constant;
-    [self.textInput invalidateIntrinsicContentSize];
+  BOOL shouldInvalidateSize = NO;
+  CGFloat widthConstant = MDCTextInputClearButtonImageSquareWidthHeight * [self clearButtonAlpha];
```

Comment:
Nit/unrelated: This is a bit surprising on first read. Would it make sense in a follow-up CR to check `clearButtonAlpha` for non-zero or zero and then make the multiplier 1/0?